src/Tree/Strategy/ORM/Closure.php

```diff
@@ -489,6 +506,10 @@ protected function setLevelFieldOnPendingNodes(ObjectManager $em)
             $uow = $em->getUnitOfWork();
 
             foreach ($this->pendingNodesLevelProcess as $node) {
+                if (get_class($node) !== $className) {
+                    // Only process nodes of the same time as the first element
```

Comment:
Did you mean "same type"?

Reply:
Yes, thanks for pointing that out. I've corrected the typo

Reply:
The commits from Oct 27 are to correct the setLevelFieldOnPendingNodes, because I realized it was also assuming that only one type of entity was being persisted. Hopefully that resolves the issue, but let me know if anything else is needed from me in order for the pull request to go through. Thanks